templates/cli/src/Client.php.twig

```diff
@@ -0,0 +1,324 @@
+<?php
+
+namespace {{ spec.title | caseUcfirst }};
+
+use Exception;
+use Utopia\CLI\Console;
+
+class Client
+{
+    const METHOD_GET = 'GET';
+    const METHOD_POST = 'POST';
+    const METHOD_PUT = 'PUT';
+    const METHOD_PATCH = 'PATCH';
+    const METHOD_DELETE = 'DELETE';
+    const METHOD_HEAD = 'HEAD';
+    const METHOD_OPTIONS = 'OPTIONS';
+    const METHOD_CONNECT = 'CONNECT';
+    const METHOD_TRACE = 'TRACE';
+
+    const USER_PREFERENCES_FILE = __DIR__."/.prefs.json";
+    const PREFERENCE_ENDPOINT = "endpoint";
+
+    /**
+     * Is Self Signed Certificates Allowed?
+     *
+     * @var bool
+     */
+    private $selfSigned = false;
+
+    /**
+     * Global Headers
+     *
+     * @var array
+     */
+    private $headers = [
+        'content-type' => '',
+        'x-sdk-version' => '{{spec.title | caseDash}}:{{ language.name | caseLower }}:{{ sdk.version }}',
+    ];
+
+    /**
+     * SDK constructor.
+     */
+    public function __construct()
+    {
+        /* Load user defaults from a json file if available
+           Else Propmt the user to enter the details
+        */
+        if (!$this->loadPreferences()) {
+            $this->promptUser();
+        }
+
+        $this
+{% for header in spec.global.headers %}
+{% if header.key != 'Mode' %}
```

Comment:
Very Appwrite specific, we should figure a better way to detect required security headers.

Reply:
We will retain this temporarily until it is removed from all the server-side SDK specs.